### PR TITLE
Fix `channels_remap` handling in `extra_envs` configuration

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

An oversight in the strip logic does not handle `channel_remaps` correctly (it assumed list of str, but we have list of dict[str, str] there). 

This input file:

```yaml
...
extra_envs:
  napari-0.5.0a2.dev751+g3b0612ec:
    channels:
    - https://github.com/napari/pins/releases/download/napari-v0.5
    channels_remap:  # THIS IS THE PROBLEM
    - src: https://github.com/napari/pins/releases/download/napari-v0.5
      dest: napari-v0.5.0a2.dev751+g3b0612ec
    specs:
    - python=3.9.*=*_cpython
    - napari=0.5.0a2.dev751+g3b0612ec
    - napari-menu=0.5.0a2.dev751+g3b0612ec
    - napari-plugin-manager
    - pyside2=*
    - conda
    - mamba
    - pip
    menu_packages:
    - napari-menu
...
```

results in the following error:

```
Traceback (most recent call last):
  File "/home/runner/micromamba/envs/build-bundle/bin/constructor", line 10, in <module>
    sys.exit(main())
  File "/home/runner/micromamba/envs/build-bundle/lib/python3.9/site-packages/constructor/main.py", line 424, in main
    main_build(dir_path, output_dir=out_dir, platform=args.platform,
  File "/home/runner/micromamba/envs/build-bundle/lib/python3.9/site-packages/constructor/main.py", line 153, in main_build
    env_config[config_key] = [val.strip() for val in value]
  File "/home/runner/micromamba/envs/build-bundle/lib/python3.9/site-packages/constructor/main.py", line 153, in <listcomp>
    env_config[config_key] = [val.strip() for val in value]
AttributeError: 'CommentedMap' object has no attribute 'strip'
```

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: constructor/main.py [hunk 1]
```diff
@@ -149,10 +149,16 @@ def main_build(dir_path, output_dir='.', platform=cc_platform,
         if env_name in ("base", "root"):
             raise ValueError(f"Environment name '{env_name}' cannot be used")
         for config_key, value in env_config.copy().items():
-            if isinstance(value, (list, tuple)):
-                env_config[config_key] = [val.strip() for val in value]
             if config_key == "environment_file":
                 env_config[config_key] = abspath(join(dir_path, value))
+            elif config_key == "channels_remap":
+                env_config[config_key] = [
+                    {"src": item["src"].strip(), "dest": "dest".strip()} for item in value
```
Comment (marcoesters) [inline constructor/main.py:156]:
```suggestion
                    {"src": item["src"].strip(), "dest": item["dest"].strip()} for item in value
```

The way this code is written, it looks like the `dest` will always be "dest". I'm surprised the tests pass - sounds like the example is maybe not as representative as they should be?

File: constructor/main.py [hunk 2]
```diff
@@ -149,10 +149,16 @@ def main_build(dir_path, output_dir='.', platform=cc_platform,
...
+                ]
+            elif isinstance(value, (list, tuple)):
+                env_config[config_key] = [val.strip() for val in value]
+            else:
+                env_config[config_key] = value
```
Comment (marcoesters) [inline constructor/main.py:161]:
```suggestion
                env_config[config_key] = value.strip()
```
Should we strip here, too?

File: CONSTRUCT.md [hunk 3]
```diff
@@ -67,7 +67,7 @@ See notes in `channels_remap` for details about local channels.
 _required:_ no<br/>
 _type:_ list<br/>
 
-A list of `src/dest` channel pairs. When building the installer, conda will
+A list of `src/dest` channel URL pairs. When building the installer, conda will
```
Comment (marcoesters) [inline CONSTRUCT.md:70]:
```suggestion
A list of `src/dest` channel/URL pairs. When building the installer, conda will
```

Looks more explicit to me.

File: constructor/construct.py [hunk 4]
```diff
@@ -42,7 +42,7 @@
 '''),
 
     ('channels_remap',         False, list, '''
-A list of `src/dest` channel pairs. When building the installer, conda will
+A list of `src/dest` channel URL pairs. When building the installer, conda will
```
Comment (marcoesters) [inline constructor/construct.py:45]:
```suggestion
A list of `src/dest` channel/URL pairs. When building the installer, conda will
```

Comment (jaimergp) [reply constructor/construct.py:45]: That makes it look like it's either a channel (name) or a URL, but I want to say "only URLs pointing to a channel". Is that clear or should I rephrase?

File: constructor/main.py (hunk 2 above)
Comment (jaimergp) [reply constructor/main.py:161]: Not guaranteed to be a `str`. We would need another clause `elif isinstance(...)`.

File: constructor/construct.py (hunk 4 above)
Comment (marcoesters) [reply constructor/construct.py:45]: Ah, I saw it as `src/dest` becomes `channel/URL`. Funny how different perceptions are. It's certainly not a blocker, just a suggestion.

**Review by marcoesters — APPROVED** [review]

**Pull request merged**